calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -171,7 +171,7 @@ func unmarshalResource(tm unversioned.TypeMetadata, b []byte) ([]unversioned.Res
 		return nil, err
 	}
 
-	if err = yaml.Unmarshal(b, unpacked); err != nil {
+	if err = yaml.UnmarshalStrict(b, unpacked); err != nil {
```

Comment:
That is a good question, and at the moment I am kind of cheating.  
The UnmarshalUnknownFieldError displays a useful message that is geared towards our requirements, and I've updated the other JSON parsing/unparsing error messages to also be geared towards our requirements (i.e. to display text in a way that makes more sense to us).

That is not my long term approach though.  I intend to (but not for v1.0.0) return the error messages in the json/yaml packages back to their more json/yamlly format and in our code we'll switch on the different error types and craft our own error message out of them.   I guess that is where the wrapped error package will come in handy - and we should take the time to understand that and start doing our error handling properly.

But for v1.0.0 I wanted something that was quick and easy to test and to implement.